### DISCUSSION TITLE
Potential for more resolution? click, double click, long press start, during long press, long press end

### DISCUSSION THREAD

**Discussion by fOmey** [jsware/able-buttons#4] [Ideas]
It would be nice to see all the event types wrapped up in a single button type, it will provide much more resolution to single buttons.

If we could detect clicks, double clicks, long press start, during long press & long press end on a single button this would allow much more flexibility..

Thoughts?

**Comment by MickeyGentry** [comment]
I have run into the same situation. The library is good unless you need to click and double click from the same button. 

Have you found any solutions to this? Obviously there are other button libraries out there. This is one but there is a lag between pressing the button and see the change, so for me that also wasn't optimal:
https://github.com/mathertel/OneButton

**Reply by JohnScottUK** [reply]
Each button type includes all the functionality from the more simple button types. For example AblePulldownCallbackDoubleClickerButton contains all the functionality available in a single button type. If you don't want callbacks, use AblePulldownDoubleClickerButton.

Check out V0.4.0 which now includes isClicked, isSingleClicked and isDoubleClicked methods. isClicked returns true if a button has been clicked, isDoubleClicked returns true if a button has been double clicked. isSingleClicked returns true if a button has been single-clicked (but has been too long for a double-click).

What I think you are after is the single-clicked vs double-clicked functions/events to detect an explicit single click vs isClicked() which returns true for the first click of a double-click.

**Comment by MickeyGentry** [comment]
Thanks!
